client/hijack.go

```diff
@@ -98,7 +98,16 @@ func (cli *Client) setupHijackConn(req *http.Request, proto string) (_ net.Conn,
 	// Server hijacks the connection, error 'connection closed' expected
 	resp, err := clientconn.Do(req)
 	if resp != nil {
-		span.SetStatus(httpconv.ClientStatus(resp.StatusCode))
+		// Very simplified variant of "httpconv.ClientStatus(resp.StatusCode))"
+		//
+		// https://github.com/open-telemetry/opentelemetry-go/blob/v1.21.0/semconv/v1.17.0/httpconv/http.go#L85-L89
+		// https://github.com/open-telemetry/opentelemetry-go/blob/v1.21.0/semconv/internal/v2/http.go#L322-L330
+		// https://github.com/open-telemetry/opentelemetry-go/blob/v1.21.0/semconv/internal/v2/http.go#L356-L404
+		code := codes.Unset
+		if resp.StatusCode >= http.StatusBadRequest {
+			code = codes.Error
+		}
+		span.SetStatus(code, "")
```

Comment:
Do we need this part if we don't have the "httpconv.ClientRequest"? (disabled that part, per suggestion above)

Reply:
Maybe outside the scope of this PR, but worth exploring updating this logic to use `http.Transport` with a custom dialer since `httputil.NewClientConn` is deprecated. The `otelhttp` library could wrap the transport in that case and maybe simplify this.

Reply:
Actually could probably be even easier to just wrap `clientconn.Do` as an `http.RoundTripper` and could use https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp#NewTransport. Not sure if that matches what the old functionality was doing.

Reply:
Playing around with that idea here https://github.com/moby/moby/pull/46920. I'm curious if that makes this clearer or cleaner.